### ISSUE TITLE
Tests don't work from external crate when defining new system of quantities.

### ISSUE THREAD

**Issue by Yoshanuikabundi** [iliekturtles/uom#98]
Hi,

I'm trying to define a new system of quantities. I've got my code to compile and run as expected (from what I can tell without tests!), but attempting to run tests on any code that uses the `system!` macro fails:
```
$ cargo test
   Compiling uomtest v0.1.0 (file:///home/yoshanuikabundi/projects/uomtest)
error[E0433]: failed to resolve. Could not find `tests` in `$crate`
  --> src/main.rs:69:1
   |
69 | / system! {
70 | |     quantities: Q {
71 | |         length: meter, L;
72 | |         mass: kilogram, M;
...  |
80 | |     }
81 | | }
   | |_^ Could not find `tests` in `$crate`
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

--- snip ---

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0433`.
error: Could not compile `uomtest`.

To learn more, run the command again with --verbose.
```
I'm using `uom` 0.19.0 and Rust 1.28.0. `uomtest` is just the mks.rs example. Oh, and I think the example is incorrect - I had to change the two `l1.get` lines to get it to compile. My code is below for reference.

```//! Example showing how to create a custom system of quantities.

#[macro_use]
extern crate uom;

use length::{foot, meter};

fn main() {
    let l1 = f32::Length::new::<meter>(100.0);

    println!(
        "{:?} {} = {:?} {}",
        l1.get(::meter),
        meter::abbreviation(),
        l1.get(::foot),
        foot::abbreviation()
    );
}

#[macro_use]
mod length {
    quantity! {
        /// Length (base unit meter, m<sup>1</sup>).
        quantity: Length; "length";
        /// Length dimension, m<sup>1</sup>.
        dimension: Q<
            P1,  // length
            Z0,  // mass
            Z0>; // time
        units {
            @meter: 1.0E0; "m", "meter", "meters";
            @foot: 3.048E-1; "ft", "foot", "feet";
        }
    }
}

#[macro_use]
mod mass {
    quantity! {
        /// Mass (base unit kilogram, kg<sup>1</sup>).
        quantity: Mass; "mass";
        /// Mass dimension, kg<sup>1</sup>.
        dimension: Q<
            Z0,  // length
            P1,  // mass
            Z0>; // time
        units {
            @kilogram: 1.0; "kg", "kilogram", "kilograms";
        }
    }
}

#[macro_use]
mod time {
    quantity! {
        /// Time (base unit second, s<sup>1</sup>).
        quantity: Time; "time";
        /// Time dimension, s<sup>1</sup>.
        dimension: Q<
            Z0,  // length
            Z0,  // mass
            P1>; // time
        units {
            @second: 1.0; "s", "second", "seconds";
        }
    }
}

system! {
    quantities: Q {
        length: meter, L;
        mass: kilogram, M;
        time: second, T;
    }

    units: U {
        mod length::Length,
        mod mass::Mass,
        mod time::Time,
    }
}

mod f32 {
    mod s {
        pub use *;
    }

    Q!(f32::s, f32);
}
```

Thanks!
 - Josh

**Comment by iliekturtles** [comment]
Thanks for the report! I was able to reproduce and pushed a temporary fix to the `dev-test-fix` branch (branched from `v0.19.0`) if you don't mind using a git dependency for the short-term. I need a bit more time to find the real fix and publish (possibly as `v0.19.1`).

Essentially the `system!` macro generates code with `#[cfg(test)]` that references code in the `uom` crate. When testing a third-party crate like the `mks` example or your code the `uom` crate itself isn't compiled with `test` so that type isn't available and the build fails.

**Comment by Yoshanuikabundi** [comment]
Thanks!
